frontend/src/bundles/talent-onboarding/store/actions.ts

```diff
@@ -9,8 +9,32 @@ const updateTalentDetails = createAsyncThunk<
     UserDetailsGeneralCustom,
     UserDetailsGeneralCustom,
     AsyncThunkConfig
->(`${sliceName}/update-talent-details`, (registerPayload) => {
-    return registerPayload;
+>(
+    `${sliceName}/update-talent-details`,
+    (registerPayload, { getState, extra }) => {
+        const { talentOnBoardingApi } = extra;
+
+        if ('badges' in registerPayload) {
+            return registerPayload;
+        }
```

Comment:
why do we need this logic?

Reply:
I forgot to write a comment that it's a temporary solution.